NeuroID/NeuroIDClass/Extensions/NIDUser.swift

```diff
@@ -94,61 +88,59 @@ public extension NeuroID {
+        var originCode = SessionOrigin.NID_ORIGIN_CODE_FAIL.rawValue
+        if validID {
+            originCode =  userGenerated ? SessionOrigin.NID_ORIGIN_CODE_CUSTOMER.rawValue : SessionOrigin.NID_ORIGIN_CODE_NID.rawValue
         }
-        let res = setGenericUserID(
-            userId: userId, type: .userID
-        ) { success in
-            if success {
-                sendOriginEvent(origin: CURRENT_ORIGIN!, originCode: CURRENT_ORIGIN_CODE!, originSessionID: userId)
-                NeuroID.userID = userId
-            }
-
-            return success
+        return SessionIDOriginalResult(origin: origin,originCode: originCode,idValue: idValue,idType: idType)
+    }
+    
+    static func setUserID(_ userId: String, _ userGenerated: Bool? = true)-> Bool{
+        let validID =  setGenericUserID(type: .userID, genericUserID: userId, userGenerated: userGenerated ?? true)
+        
+        if(!validID){
+            return false
         }
-
-        return res
+        
+        NeuroID.userID = userId
+        return true
     }
-
+    
     static func getUserID() -> String {
         return NeuroID.userID ?? ""
     }
-
+    
+    
     static func getRegisteredUserID() -> String {
         return NeuroID.registeredUserID
     }
-
+    
     static func setRegisteredUserID(_ registeredUserID: String) -> Bool {
         if !NeuroID.registeredUserID.isEmpty, registeredUserID != NeuroID.registeredUserID {
             NeuroID.saveEventToLocalDataStore(NIDEvent(level: "warn", m: "Multiple Registered User Id Attempts"))
             NIDLog.e("Multiple Registered UserID Attempt: Only 1 Registered UserID can be set per session")
             return false
         }
-
-        let res = setGenericUserID(
-            userId: registeredUserID, type: .registeredUserID
-        ) { success in
-            if success {
-                NeuroID.registeredUserID = registeredUserID
-            }
-            CURRENT_ORIGIN = SessionOrigin.NID_ORIGIN_CUSTOMER_SET.rawValue
-            CURRENT_ORIGIN_CODE = SessionOrigin.NID_ORIGIN_CODE_CUSTOMER.rawValue
-            sendOriginEvent(origin: CURRENT_ORIGIN!, originCode: CURRENT_ORIGIN_CODE!, originSessionID: registeredUserID)
-
-            return success
+        
+        let validID = setGenericUserID(type: .registeredUserID, genericUserID: registeredUserID)
+        
+        if(!validID){
+            return false
         }
-
-        return res
+        
+        NeuroID.registeredUserID = registeredUserID
+        return true
     }
-
+    
     /**
-        This should be called the moment a user trys to login. Returns true always
-        @param {String} [attemptedRegisteredUserId] - an optional identifier for the login
+     This should be called the moment a user trys to login. Returns true always
+     @param {String} [attemptedRegisteredUserId] - an optional identifier for the login
```

Comment:
I don't know if it was included in the ticket but in the Android PR we utilize the setGenericUserID method in the attemptedLogin method so we can capture that origin event as well